src/main/java/com/pokedex/springmvc/controller/UserController.java

```diff
@@ -0,0 +1,126 @@
+package com.pokedex.springmvc.controller;
+ 
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import com.pokedex.springmvc.model.User;
+import com.pokedex.springmvc.service.UserService;
+ 
+@RestController
+public class UserController {
+ 
+    @Autowired
+    UserService userService; 
+ 
+    
+    //-------------------Retrieve All Users--------------------------------------------------------
+     
+    @RequestMapping(value = "/user/", method = RequestMethod.GET)
+    public ResponseEntity<List<User>> listAllUsers() {
```

Comment:
Change ResponseEntity<List<User>> to ResponseEntity<?>. This goes for all rest endpoints. So that the rest end points are not so restrictive in what they accept.